poll/async/buildUpdatedClosingTime.js

```diff
@@ -1,46 +1,45 @@
 const pull = require('pull-stream')
 const pullAsync = require('pull-async')
-const { heads } = require('ssb-sort')
 const { isMsg } = require('ssb-ref')
 const { isPoll, isPollUpdate, versionStrings: {V1_SCHEMA_VERSION_STRING} } = require('ssb-poll-schema')
+const getMessages = require('./getMessages')
+const getHeads = require('../sync/getHeads')
 
 module.exports = function (server) {
   return function UpdatedClosingTime ({ poll, mentions, recps, closesAt }, cb) {
     if (!isPoll(poll) && !isMsg(poll)) return cb(new Error('UpdatedClosingTime factory expects a valid poll'))
 
-    // NOTE - getPoll has to be required here to avoid circular deps
-    const getPoll = require('../../poll/async/get')(server)
-
-    pull(
-      pullAsync(cb => {
-        getPoll(typeof poll === 'string' ? poll : poll.key, cb)
-      }),
-      pull.map(build),
-      pull.drain(update => {
-        if (!isPollUpdate(update)) return cb(isPollUpdate.errors)
-
-        cb(null, update)
-      })
-    )
-
-    function build (poll) {
+    function build ({ key, heads }) {
       // NOTE - poll here is a decorated poll
       const content = {
         type: 'poll-update',
         version: V1_SCHEMA_VERSION_STRING,
-        root: poll.key,
-        closesAt
+        closesAt,
+        root: key,
+        branch: heads
       }
 
-      content.branch = heads(poll.positions.concat(poll))
-
       if (recps) content.recps = recps
       if (mentions) content.mentions = mentions
-      if (poll.channel) content.channel = poll.channel
```

Comment:
I dropped support for channel here ... doesn't seem all that imporant for poll-updates..